kevm-pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -39,30 +40,92 @@ class Method:
         sort: KSort
         arg_names: tuple[str, ...]
         arg_types: tuple[str, ...]
+        contract: Contract
```

Comment:
Hmmm, it seems strange to me that we have to pass in the `contract` to the `Method`, but `Method` is a nested class in `Contract`. I guess it doesn't matter type-wise.

But if we're going to do this, I guess we shouldn't also store `contract_name` separately, instead just use `self.contract.name` everywhere.

Reply:
I changed it so we pass in the contract name and digest instead of a reference to the whole contract.